src/common/PreviewUtils.ts

```diff
@@ -128,4 +129,44 @@ export class PreviewUtils {
             return undefined;
         }
     }
+
+    /**
+     * For the new Local Development Preview approach where a new local dev server is being used,
+     * the local dev server needs to be configured to use a web socket url for local previewing.
+     * This method generates an appropriate web socket url to be used by the local dev server.
+     *
+     * @param platform a platform flag (desktop , ios, android).
+     * @param port the port number that the local dev server is configured to use.
+     * @returns A string representing a web socket url to be used by the local dev server.
+     */
+    public static generateWebSocketUrlForLocalDevServer(platform: string, port: string): string {
+        /*
+          - For desktop browsers other than Safari, local development use cases will target ws://localhost:<port> connections to the local dev server
+          - For the Safari desktop browser, target wss://localhost:<port>
+            
+          - For mobile (webview in native apps), local development use cases will target:
+            - iOS: wss://localhost:<port>
+            - Android: wss://10.0.2.2:<port>
+        */
+
+        if (CommandLineUtils.platformFlagIsIOS(platform)) {
+            return `wss://localhost:${port}`;
+        }
+
+        if (CommandLineUtils.platformFlagIsAndroid(platform)) {
+            return `wss://10.0.2.2:${port}`;
+        }
+
+        if (process.platform !== 'darwin') {
+            return `ws://localhost:${port}`; // cannot be Safari since it is only available on Mac
+        }
+
+        // Check to see if the default browser is Safari
```

Comment:
Nit: Can you preface this comment with something like "macOS use case:"? Just to make it clear that, in this block, we're working in a macOS environment.